BuffTrigger.lua

```diff
@@ -52,6 +52,9 @@ CanHaveTooltip(data)
 GetNameAndIcon(data)
   Returns the name and icon to show in the options
 
+GetAdditionalProperties(data)
+  Returns the a tooltip for the additional properties
+
 ]]--
```

Comment:
So with this patch, the bufftrigger stores a bit of extra information to the trigger state, namely spellId and the caster name. The caster name was asked for in http://www.wowace.com/addons/weakauras-2/tickets/291-caster-dynamic-text/ . The spellId could be useful for writing custom texts.

This patch thus does two things: a) Add infrastructure to allow for additional states, and b) modifies the buff trigger to actually store the caster and spellId.